checks/libs/wmi/sampler.py

```diff
@@ -47,18 +47,57 @@ def get(self, key):
         return super(CaseInsensitiveDict, self).get(key.lower())
 
 
+class ProviderArchitectureMeta(type):
+    """
+    Metaclass for ProviderArchitecture.
+    """
+    def __contains__(cls, provider):
+        """
+        Support `Enum` style `contains`.
+        """
+        return provider in cls._AVAILABLE_PROVIDER_ARCHITECTURES
+
+
+class ProviderArchitecture(object):
+    """
+    Enumerate WMI Provider Architectures.
+    """
+    __metaclass__ = ProviderArchitectureMeta
+
+    # Available Provider Architecture(s)
+    DEFAULT = 0
+    _32BIT = 32
+    _64BIT = 64
+    _AVAILABLE_PROVIDER_ARCHITECTURES = frozenset([DEFAULT, _32BIT, _64BIT])
+
+
 class WMISampler(object):
     """
     WMI Sampler.
     """
+    # Properties
+    _provider = None
+    _formatted_filters = None
+
+    # Type resolution state
+    _property_counter_types = None
+
+    # Samples
+    _current_sample = None
+    _previous_sample = None
+
+    # Sampling state
+    _sampling = False
```

Comment:
The only thing that I don't like with this (and part of the reason why I did it), is that it doesn't look great when using a Python property, i.e.

```python
class WMISampler(object):
    """
    An OK looking WMISampler 🌧
    """
    def __init__(self, provider):
        self._provider = None
        self.provider = provider
```
```python
class WMISampler(object):
    """
    A good looking WMISampler ☀😎
    """
    _provider = None

    def __init__(self, provider):
        self.provider = provider
```